vector_store.go

```diff
@@ -0,0 +1,349 @@
+	ID      string `json:"id"`
+	Object  string `json:"object"`
+	Deleted bool   `json:"deleted"`
+
+	httpHeader
+}
+
+type VectorStoreFile struct {
+	ID            string `json:"id"`
+	Object        string `json:"object"`
+	CreatedAt     int64  `json:"created_at"`
+	VectorStoreID string `json:"vector_store_id"`
+	UsageBytes    int    `json:"usage_bytes"`
+	Status        string `json:"status"`
+
+	httpHeader
+}
+
+type VectorStoreFileRequest struct {
+	FileID string `json:"file_id"`
+}
+
+type VectorStoreFilesList struct {
+	VectorStoreFiles []VectorStoreFile `json:"data"`
+
+	httpHeader
+}
+
+type VectorStoreFileBatch struct {
+	ID            string               `json:"id"`
+	Object        string               `json:"object"`
+	CreatedAt     int64                `json:"created_at"`
+	VectorStoreID string               `json:"vector_store_id"`
+	Status        string               `json:"status"`
+	FileCounts    VectorStoreFileCount `json:"file_counts"`
+
+	httpHeader
+}
+
+type VectorStoreFileBatchRequest struct {
+	FileIDs []string `json:"file_ids"`
+}
+
+// CreateVectorStore creates a new vector store.
+func (c *Client) CreateVectorStore(ctx context.Context, request VectorStoreRequest) (response VectorStore, err error) {
+	req, _ := c.newRequest(ctx, http.MethodPost, c.fullURL(vectorStoresSuffix), withBody(request),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// RetrieveVectorStore retrieves an vector store.
+func (c *Client) RetrieveVectorStore(
+	ctx context.Context,
+	vectorStoreID string,
+) (response VectorStore, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s", vectorStoresSuffix, vectorStoreID)
+	req, _ := c.newRequest(ctx, http.MethodGet, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// ModifyVectorStore modifies a vector store.
+func (c *Client) ModifyVectorStore(
+	ctx context.Context,
+	vectorStoreID string,
+	request VectorStoreRequest,
+) (response VectorStore, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s", vectorStoresSuffix, vectorStoreID)
+	req, _ := c.newRequest(ctx, http.MethodPost, c.fullURL(urlSuffix), withBody(request),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// DeleteVectorStore deletes an vector store.
+func (c *Client) DeleteVectorStore(
+	ctx context.Context,
+	vectorStoreID string,
+) (response VectorStoreDeleteResponse, err error) {
+	urlSuffix := fmt.Sprintf("%s/%s", vectorStoresSuffix, vectorStoreID)
+	req, _ := c.newRequest(ctx, http.MethodDelete, c.fullURL(urlSuffix),
+		withBetaAssistantVersion(c.config.AssistantVersion))
+
+	err = c.sendRequest(req, &response)
+	return
+}
+
+// ListVectorStores Lists the currently available vector store.
+func (c *Client) ListVectorStores(
+	ctx context.Context,
+	limit *int,
+	order *string,
+	after *string,
+	before *string,
```

Comment:
Could we please use [Pagination](https://github.com/sashabaranov/go-openai/blob/211cb49fc22766f4174fef15301c4d39aef609d3/run.go#L210) struct here?

Reply:
We might as well put pagination into a separate file and provide a `NoPagination = Pagination{}` const